FurkanGozukara/SUPIR
# Dev

### Fixes for model loading and download:

- Run download_models.py is optional now. If run, the models will be saved in the default 'models' folder within the project root path. Checkpoints will be stored in the 'checkpoints' subfolder within the 'models' directory.
- The code is already prepared so that during execution it can carry out all necessary downloads in the default 'models' folder if it does not exist in the root directory.
- The options yaml file is no longer used to reference checkpoint names.
- With the exception of the automatic download script, all code will search for model names and parameterized checkpoints in the CKPT_PTH.py file. This file will receive the default 'models' base folder in the 'model_path' variable when the 'args.ckpt_dir' argument is not provided. I created a second variable 'model_path2' which is optional if you want to use a second path for the model (e.g. kaggle temp folder). This variable is defined when the 'args.ckpt_dir2' argument is entered. If the argument is not passed, it will receive the same value as 'model_path'.

**How to test:**

**Test 1**: (Do not enter any arguments for ckpt_dir and ckpt_dir2) - Remove the 'models' folder if it exists from the project root path. Just run the code and all the necessary templates will be automatically downloaded during execution.
**Test 2**: (Enter a different path for the ckpt_dir argument) All files will be downloaded in this informed path, if they already exist, they will be used, however the existing folder structure must match the pattern of how they are originally downloaded.
**Test 3**: (Enter a different path for the ckpt_dir2 argument) The laion clip model will be downloaded to this new path.
**Test 4**: - Delete the model folder in the project root and run 'download_models.py', it should download all models back to the default folder.

**Structure model folders:**

---> **models**
| ------**checkpoints** (Here is ckpt supir and others safetensor or ckpt files)
|
|**others models folders/subfolders** (it is like model_id e.g: 'laion/CLIP-ViT-bigG-14-laion2B-39B-b160k' like subfolder)

### Fix for FP16 on Kaggle
For Kaggle you need start with follow params:

For FP16:
!python gradio_demo.py --loading_half_params --use_tile_vae --load_4bit_llava --ckpt_dir /kaggle/temp/models

For FP8:
!python gradio_demo.py --fp8  --use_tile_vae --load_4bit_llava --ckpt_dir /kaggle/temp/models

The best way to test is to use the notebook that I have already made available as there are also some dependency adjustments for the installation. For Kaggle, a separate requirements file called requirements_kaggle.txt was created.

Other changes:
FP8: --fp8 new parameter to use FP8, do not require --loading_half_params together
